tree/ntuple/src/RNTupleProcessor.cxx

```diff
@@ -477,61 +400,39 @@ void ROOT::Experimental::RNTupleJoinProcessor::SetModel(std::unique_ptr<ROOT::RN
+   fProtoModel->AddField(std::move(auxParentField));
 
    for (const auto &field : subFields) {
-      fModel->RegisterSubfield(field->GetQualifiedFieldName());
+      fProtoModel->RegisterSubfield(field->GetQualifiedFieldName());
 
       if (field->GetTypeName() == "RAuxiliaryProcessorField") {
          for (const auto &auxSubField : field->GetConstSubfields()) {
-            fModel->RegisterSubfield(auxSubField->GetQualifiedFieldName());
+            fProtoModel->RegisterSubfield(auxSubField->GetQualifiedFieldName());
          }
       }
    }
-
-   // If the model has a default entry, adopt its value pointers. This way, the pointers returned by
-   // RNTupleModel::MakeField can be used in the processor loop to access the corresponding field values.
-   if (!fModel->IsBare() && !auxModel->IsBare()) {
-      const auto &auxDefaultEntry = auxModel->GetDefaultEntry();
-      auto &joinDefaultEntry = fModel->GetDefaultEntry();
-      for (const auto &fieldName : auxModel->GetFieldNames()) {
-         auto valuePtr = auxDefaultEntry.GetPtr<void>(fieldName);
-         joinDefaultEntry.BindValue(fAuxiliaryProcessor->GetProcessorName() + "." + fieldName, valuePtr);
-      }
-   }
-
-   fModel->Freeze();
 }
 
-void ROOT::Experimental::RNTupleJoinProcessor::SetEntryPointers(const ROOT::REntry &entry,
-                                                                std::string_view fieldNamePrefix)
+void ROOT::Experimental::RNTupleJoinProcessor::SetAuxiliaryFieldValidity(bool isValid)
 {
-   for (const auto &value : *fEntry) {
-      std::string fieldName = value.GetField().GetQualifiedFieldName();
-      auto valuePtr = fieldNamePrefix.empty() ? entry.GetPtr<void>(fieldName)
-                                              : entry.GetPtr<void>(std::string(fieldNamePrefix) + "." + fieldName);
-
-      fEntry->BindValue(fieldName, valuePtr);
+   for (const auto &token : fAuxiliaryFieldTokens) {
+      fEntry->SetFieldValidity(token, isValid);
    }
-
-   fPrimaryProcessor->SetEntryPointers(*fEntry);
-   fAuxiliaryProcessor->SetEntryPointers(*fEntry, fAuxiliaryProcessor->GetProcessorName());
 }
 
 ROOT::NTupleSize_t ROOT::Experimental::RNTupleJoinProcessor::LoadEntry(ROOT::NTupleSize_t entryNumber)
 {
+   EnsureEntryIsFrozen();
+
    if (fPrimaryProcessor->LoadEntry(entryNumber) == kInvalidNTupleIndex)
       return kInvalidNTupleIndex;
 
    fCurrentEntryNumber = entryNumber;
    fNEntriesProcessed++;
 
    if (!fJoinTable) {
-      if (fAuxiliaryProcessor->LoadEntry(entryNumber) == kInvalidNTupleIndex) {
-         throw RException(R__FAIL("entry " + std::to_string(entryNumber) +
-                                  " in the primary processor has no corresponding entry in auxiliary processor \"" +
-                                  fAuxiliaryProcessor->GetProcessorName() + "\""));
-      }
+      if (fAuxiliaryProcessor->LoadEntry(entryNumber) == kInvalidNTupleIndex)
+         SetAuxiliaryFieldValidity(false);
```

Comment:
Where is the inverse happening, ie where are we setting fields as valid if the next entry exists again?